src/py/mains/consensus.py

```diff
@@ -13,6 +13,27 @@
 falcon.free_consensus_data.argtypes = [ POINTER(falcon_kit.ConsensusData) ]
 
 
+def get_longest_reads(seqs, max_n_read, max_cov_aln, sort=True):
+    # including the sort kwarg allows us to avoid a redundant sort
+    # in get_consensus_trimmed()
+    if sort: 
+        seqs = seqs[:1] + sorted(seqs[1:], key=lambda x: -len(x))
+
+    longest_n_reads = max_n_read
+    if max_cov_aln > 0:
+        longest_n_reads = 0
```

Comment:
longest_n_reads = 1

I don't understand your comment below. The original code did this:
```
if len(seqs) > max_n_read:
    seqs = seqs[:max_n_read]
```
So max_n_read refers to the entire length of `seq`. But in your case, if `len(seq)==1`, then `longest_n_reads==0` (if `max_cov_aln>0`). Your for-loop seems to be off-by-1 always. I think you should also move the `if` ahead of the increment.